megatron/model/transformer.py

```diff
@@ -594,6 +597,27 @@ def forward(self, inputs, **kwargs):
 class ParallelTransformer(MegatronModule):
     """Transformer class."""
 
+    @staticmethod
+    def _build_alibi_tensor(max_seq_len, num_attention_heads, batch_size):
+        # Based on https://github.com/ofirpress/attention_with_linear_biases/blob/a35aaca144e0eb6b789dfcb46784c4b8e31b7983/fairseq/models/transformer.py#L742
+        """Returns tensor shaped (batch_size * num_attention_heads, 1, max_seq_len)"""
+        def get_slopes(n):
+            def get_slopes_power_of_2(n):
+                start = (2 ** (-2 ** -(math.log2(n) - 3)))
+                ratio = start
+                return [start * ratio ** i for i in range(n)]
+
+            if math.log2(n).is_integer():
+                return get_slopes_power_of_2(n)
+            else:
+                closest_power_of_2 = 2 ** math.floor(math.log2(n))
+                return get_slopes_power_of_2(closest_power_of_2) + get_slopes(2 * closest_power_of_2)[0::2][
+                                                                   :n - closest_power_of_2]
+        slopes = torch.Tensor(get_slopes(num_attention_heads))
+        alibi = slopes.unsqueeze(1).unsqueeze(1) * torch.arange(max_seq_len).unsqueeze(0).unsqueeze(0).expand(num_attention_heads, -1, -1)
+        alibi = alibi.view(num_attention_heads, 1, max_seq_len).repeat(batch_size, 1, 1)
```

Comment:
Isn't alibi already the correct format? Also why not expand?

```suggestion
        alibi = alibi.repeat(batch_size, 1, 1)
```

Reply:
Ya that view is not necessary, thanks!

About repeat vs expand-
Expand can only be used to expand dims that are of size 1, so it can't be used here unfortunately :(